src/lib/utils/letterDataUtils.test.ts

```diff
@@ -93,3 +101,173 @@ describe('makeWordUsed', () => {
 		});
 	});
 });
+
+describe('provideHint', () => {
+	it('should populate a hint into playedLetters', () => {
+		let playedLetters: LetterData[] = [
+			{ value: 'h', state: 'firstLetter' },
+			{ value: 'e', state: 'hint' },
+			{ value: 'o', state: 'inPlay' }
+		];
+
+		let unplayedLetters: LetterData[] = [
+			{ value: 'l', state: 'inPlay' },
+			{ value: 'l', state: 'inPlay' }
+		];
+
+		[playedLetters, unplayedLetters] = provideHint(playedLetters, unplayedLetters, 'hello');
+
+		const expectedPlayedLetters: LetterData[] = [
+			{ value: 'h', state: 'firstLetter' },
+			{ value: 'e', state: 'hint' },
+			{ value: 'l', state: 'hint' }
+		];
+
+		const expectedUnplayedLetters: LetterData[] = [
+			{ value: 'l', state: 'inPlay' },
+			{ value: 'o', state: 'inPlay' }
+		];
+
+		expect(playedLetters).toEqual(expectedPlayedLetters);
+		expectArrayEquivalence(unplayedLetters, expectedUnplayedLetters);
+	});
+	describe('given a previous word has been solved', () => {
+		it('should still populate a hint into playedLetters', () => {
+			let playedLetters: LetterData[] = [
+				{ value: 'y', state: 'used' },
+				{ value: 'a', state: 'used' },
+				{ value: 'h', state: 'used' },
+				{ value: 'o', state: 'used' },
+				{ value: 'o', state: 'used' },
+				{ value: 'h', state: 'firstLetter' },
+				{ value: 'e', state: 'hint' },
+				{ value: 'o', state: 'inPlay' }
+			];
+
+			let unplayedLetters: LetterData[] = [
+				{ value: 'l', state: 'inPlay' },
+				{ value: 'l', state: 'inPlay' }
+			];
+
+			[playedLetters, unplayedLetters] = provideHint(playedLetters, unplayedLetters, 'hello');
+
+			const expectedPlayedLetters: LetterData[] = [
+				{ value: 'y', state: 'used' },
+				{ value: 'a', state: 'used' },
+				{ value: 'h', state: 'used' },
+				{ value: 'o', state: 'used' },
+				{ value: 'o', state: 'used' },
+				{ value: 'h', state: 'firstLetter' },
+				{ value: 'e', state: 'hint' },
+				{ value: 'l', state: 'hint' }
+			];
+
+			const expectedUnplayedLetters: LetterData[] = [
+				{ value: 'l', state: 'inPlay' },
+				{ value: 'o', state: 'inPlay' }
+			];
+
+			expect(playedLetters).toEqual(expectedPlayedLetters);
+			expectArrayEquivalence(unplayedLetters, expectedUnplayedLetters);
+		});
+	});
+});
+
+describe('resetBoard', () => {
+	it('should return both boards with all "inPlay" letters in the unplayedLetters board', () => {
+		let playedLetters: LetterData[] = [
+			{ value: 'h', state: 'firstLetter' },
+			{ value: 'e', state: 'inPlay' },
+			{ value: 'o', state: 'inPlay' }
```

Comment:
this is just me musing aloud, dont need to change anything necessarily, but it seems like there's a missing `state` here. instead of `inPlay` I wonder if `played` and `unplayed` make more sense

Reply:
Noted. I might make a ticket to reanalyze the state names now that the game has changed a bit.